programs/uxd/src/instructions/mint_with_router.rs

```diff
@@ -0,0 +1,437 @@
+use anchor_lang::prelude::*;
+use anchor_spl::associated_token::AssociatedToken;
+use anchor_spl::token::Mint;
+use anchor_spl::token::Token;
+use anchor_spl::token::TokenAccount;
+
+use crate::error::UxdError;
+use crate::state::controller::Controller;
+use crate::state::credix_lp_depository::CredixLpDepository;
+use crate::state::identity_depository::IdentityDepository;
+use crate::state::mercurial_vault_depository::MercurialVaultDepository;
+use crate::utils::calculate_depositories_mint_collateral_amount;
+use crate::utils::calculate_depositories_mint_collateral_amount::DepositoryInfoForMintCollateralAmount;
+use crate::utils::calculate_depositories_target_redeemable_amount;
+use crate::utils::calculate_depositories_target_redeemable_amount::DepositoryInfoForTargetRedeemableAmount;
+use crate::utils::validate_collateral_amount;
+use crate::validate_is_program_frozen;
+use crate::CONTROLLER_NAMESPACE;
+use crate::CREDIX_LP_DEPOSITORY_NAMESPACE;
+use crate::CREDIX_LP_EXTERNAL_PASS_NAMESPACE;
+use crate::IDENTITY_DEPOSITORY_COLLATERAL_NAMESPACE;
+use crate::IDENTITY_DEPOSITORY_NAMESPACE;
+use crate::MERCURIAL_VAULT_DEPOSITORY_LP_TOKEN_VAULT_NAMESPACE;
+use crate::MERCURIAL_VAULT_DEPOSITORY_NAMESPACE;
+
+#[derive(Accounts)]
+pub struct MintWithRouter<'info> {
+    /// #1
+    pub user: Signer<'info>,
+
+    /// #2
+    #[account(mut)]
+    pub payer: Signer<'info>,
+
+    /// #3
+    #[account(
+        mut,
+        seeds = [CONTROLLER_NAMESPACE],
+        bump = controller.load()?.bump,
+        constraint = controller.load()?.registered_mercurial_vault_depositories[0] == mercurial_vault_depository_0.key() @UxdError::InvalidDepository,
+        constraint = controller.load()?.registered_credix_lp_depositories[0] == credix_lp_depository_0.key() @UxdError::InvalidDepository,
```

Comment:
Again not really for this PR, but if we don't have a streamlined way to handle all these, we can just store them as variables so at least we name them proper (with the collateral or differentiation) and we can use the more anchor-idiomatic "has_one"

Reply:
Yeah I agree, I think you explained this to me in a voice chat a few weeks back. I agree this would be good, I only left it that way because changing that would require a lot more DIFF and complexify the overall PRs.

Definitly willing to do a refactoring after that tho if you think that's appropriate!

Reply:
Discussed offline, merging for now, with intention to fix later